src/net.cpp

```diff
@@ -2861,3 +2861,23 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(const CNode& node, const std::string& msg_type, const Span<const unsigned char>& data, bool is_incoming)
+{
+    auto time = GetTime<std::chrono::microseconds>();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node.addr.ToString();
+    fs::create_directories(base_path);
+
+    fs::path path = base_path / (is_incoming ? "msgs_recv.dat" : "msgs_sent.dat");
+    CAutoFile f(fsbridge::fopen(path, "ab"), SER_DISK, CLIENT_VERSION);
+
+    f.write((const char*)&time, sizeof(time));
```

Comment:
This way of serialization doesn't seem to be endian-agnostic, i.e. it depends on the architecture whether this is written out in big or little endian format. In the parsing script you assume both time and data size are stored in little endian, hence this needs to be enforced here. I think `ser_writedata32` (see `serialization.h`) can be used for this.
It's a pity though that the CI tests don't fail, I think this is because we don't have any testing instance where the architecture uses big endian byte order :/

Reply:
Thanks for catching this!  When fixing it I considered if it was better to instead change the test to follow the system endianness (with python's `sys.byteorder`).  I think fixing a particular endianness for the .dat files, as temporary as they are, is better.  Someone might generate them on a BE system and then parse them on a LE system one day...